scout/config.py

```diff
@@ -0,0 +1,150 @@
+from __future__ import annotations
+import yaml
+import copy
+from pathlib import Path
+from jsonschema import validate
+from scout.constants import FilePaths as fp
+
+
+class Config:
+    """Methods to handle yml configuration files, including writing argparse arguments based on
+        the schema (supporting_data/config_schema.yml) and enabling validation of inputted
+        configuration files.
+
+    Attributes:
+        parser (argparse.ArgumentParser): Argument parser
+        schema_file (Pathlib.filepath): Location of the configuration schema yml
+        schema_data (dict): Parsed schema data
+        key (str): Workflow step on which to validate and set arguments {ecm_prep, run}
```

Comment:
Should there be a definition for the `cli_args` attribute?

Reply:
Yes, updated.